docs/developer-guides/merkle-root-construction.md

```diff
@@ -2,10 +2,45 @@
 
 ---
 
-#### Merkle Trees
+## Merkle Trees
 
-The merkle root is constructed using all the TXIDs of transactions in
-this block, but first the TXIDs are placed in order as required by the
+
+A merkle tree is a hash-based data structure that summarizes all the transactions in a block. It can be used to quickly and efficiently verify whether or not a transaction is included in a block. 
```

Comment:
It is true that Merkle trees are used to summarize all transactions in a block, but this line almost sounds like  transactions are a fundamental part of a merkle tree. They aren't - any type of data can be stored in a merkle tree.

Perhaps we should split this into two separate points, a high level conceptual explaination, and then a concrete example. Something along the lines of:

> A merkle tree is a data structure that can be used to store hashes of  arbitrary data

follwed by something like

> In the context of decred, this might be transaction data, or ticket data.

Reply:
Agreed. Have separated content into two sections: 'Merkle Trees' (general definition) and 'Merkle Trees in Decred' (implementation details in Decred).